src/components/login/LoginComponent.jsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable react/no-unescaped-entities */
+import { Link } from 'react-router-dom'
+import React, { Component } from 'react'
+import { users } from '../../utils/GeneralConstants'
+import GeneralInputComponent from '../layouts/GeneralInputComponent.jsx'
+
+export class LoginComponent extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      email: '',
+      password: '',
+      message: '',
+      isSuccess: true
+    }
+  }
+
+  resetAlertMessage = () => {
+    setTimeout(() => {
+      this.setState({
+        message: ''
+      })
+    }, 3000)
+  }
+
+  clearFormData = () => {
+    this.setState({
+      password: '',
+      email: ''
+    })
+  }
+
+  onChange = e => {
+    const { target: { value, name } } = e
+    this.setState({
+      [name]: value
+    })
+  }
+
+  onClickSubmit = () => {
+    const { email, password } = this.state
+    const userFound = users.findIndex(
+      user => user.email === email && user.password === password
+    )
+
+    if (userFound !== -1) {
+      this.props.history.push('/main', { email: this.state.email })
+    } else {
+      this.setState({
+        message: 'Login Failed, Please try with different credentials',
+        isSuccess: false
+      })
+      this.resetAlertMessage()
+      this.clearFormData()
+    }
+  }
+
+  render () {
+    return (
+      <>
+        <div className="form-group">
+          Welcome to Secret Santa App
+        </div>
+        {this.state.message &&
+          (this.state.isSuccess ? (
+            <div className="alert alert-success" role="alert">
+              {this.state.message}
+            </div>
+          ) : (
+            <div className="alert alert-danger" role="alert">
+              {this.state.message}
+            </div>
+          ))}
+        <form>
+          <GeneralInputComponent
```

Comment:
The `General` prefix and the `Component` suffix are extra fluff that might not be needed. Consider removing it.